lib/warder/coffee_lint_runner.rb

```diff
@@ -0,0 +1,91 @@
+require 'coffeelint'
+
+module Warder
+  # responsible for run coffeescript style guide validation
+  class CoffeeLintRunner < Runner
+    CLI_OPTION = 'g'
+    CLI_FULL_OPTION = 'coffee-lint'
+    DESCRIPTION = 'Run coffeescript style guide validation'
+    COMMAND_NAME = 'coffeelint'
+    FAILURE_REGEXP = /(?<issues>(WARN|ERROR))/
+
+    private
+
+    def run_command
+      @stdout.puts(exec_msg) unless quiet?
+      lint_results.each do |filename, result|
+        next if result.empty?
+        yield "\n#{filename}\n#{MessageFormatter.new(result).format}"
+      end
+      @stdout.puts(stats_msg) if stats?
+    end
+
+    def lint_results
+      lint_results = {}
+      @options.files.split(' ').each do |filename|
+        next if filename.length == 0
+        lint_results.merge!(LintRunner.new(filename).result)
+      end
+      lint_results
+    end
+
+    def number_of_issues(line)
+      self.class::FAILURE_REGEXP.match(line) ? 1 : 0
+    end
+
+    # responsible for coffeelint results
+    class LintRunner
+      def initialize(file_or_path)
+        @file_or_path = file_or_path
+      end
+
+      def result
+        if Dir.exist?(@file_or_path)
+          Coffeelint.lint_dir(@file_or_path)
+        else
+          { filename => Coffeelint.lint_file(@file_or_path) }
+        end
+      end
+    end
+
+    # responsible for human readable result of coffeelint validation
+    class MessageFormatter
+      def initialize(result)
+        @result = result
+      end
+
+      def format
+        @result.map do |result|
+          MessageLineFormatter.new(result).format
+        end.join("\n")
+      end
+    end
+
+    # responsible for human readable error message representation
+    class MessageLineFormatter
+      MESSAGE_KEY = 'message'.freeze
+      CONTEXT_KEY = 'context'.freeze
+      LEVEL_KEY = 'level'.freeze
+      LINE_NUMBER_KEY = 'lineNumber'.freeze
+
+      def initialize(result)
+        @result = result
+      end
+
+      def format
+        msg = [@result[MESSAGE_KEY], @result[CONTEXT_KEY]].join('. ')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.